pandas/core/frame.py

```diff
@@ -3482,11 +3482,17 @@ def __getitem__(self, key):
         if is_hashable(key) and not is_iterator(key):
             # is_iterator to exclude generator e.g. test_getitem_listlike
             # shortcut if the key is in columns
-            if self.columns.is_unique and key in self.columns:
-                if isinstance(self.columns, MultiIndex):
-                    return self._getitem_multilevel(key)
+            is_mi = isinstance(self.columns, MultiIndex)
+            # GH#45316 Return view if key is not duplicated
+            if not is_mi and (
+                self.columns.is_unique
```

Comment:
umm why can't this be just
```not is_mi and key in self.columns``` ?

Reply:
Because this works only if it returns a series. If self.columns contains key twice, this fails.

Reply:
Shouldn't `key in self.columns.drop_duplicates(keep=False)` cover the `is_unique and key in self.columns` case?

Reply:
I am pretty sure I did that for performance reasons

Reply:
Yep already did. Took me way to long to remember that for it being obvious :)